feat(tree-view): initial design guidance (#24)
vmware-clarity/clarity-guidance: 1037-treeview-design-guidance.md

---
category: component design guidance
state: draft
created: 2024-10-17
updated: 2024-10-21
---

# Treeview Component

> [Treeview](https://clarity.design/documentation/treeview) is a hierarchical list of items that have a parent-child relationship where children can be toggled into view by expanding or collapsing their parent item.

Reviewer: ```suggestion
The [tree view](https://clarity.design/documentation/tree-view) component presents a hierarchical list of items that have a parent-child relationship where children can be toggled into view by expanding or collapsing their parent item.
```

## Guidance

- **Must** use a tree view to display and work with large amounts of hierarchical information.

> - **Must** NOT use too many levels unless it is for user-generated information. Instead consolidate the layers if possible.

Reviewer: ```suggestion
- **Must not** use too many levels unless it is for user-generated information. Instead, consolidate the layers if possible.
```

> - **Must** NOT use for only one level of nested information. Use an `accordion` instead that go one level deep.

Reviewer: ```suggestion
- **Must not** use for only one level of nested information. Use an [accordion](https://clarity.design/documentation/accordion) instead.
```

Reviewer: I wonder if this one should be a "should not" instead because our documentation has an example with only one nested level.

> ## Icon Tree

Reviewer: ```suggestion
## Icon Tree

```

nit: missing blank line

> - **Should** maintain consistency for icons. If icons needs to be added to a tree view, apply the icons on all items. Avoid icons to only some items in a tree view.

Reviewer: ```suggestion
- **Must** maintain consistency for icons. If icons need to be added to a tree view, apply the icons on all items. **Must not** to only some items in a tree view.
```

The website says:

>Do not alternate between using and not using icons within the same tree.

- **Should** use neutral colors to avoid overloading the tree and related components with colors and shapes.

> - Use Icons in order to visually represent the type of items in a tree.

Reviewer: ```suggestion
- **Should** use icons in order to visually represent the type of items in a tree.
```

> ## Checkbox Tree

Reviewer: ```suggestion
## Checkbox Tree

```

nit: missing blank line

> - **Must** avoid using icons and checkboxes at the same time. It would result in large indentation and loss of space.

Reviewer: ```suggestion
- **Must not** use icons and checkboxes at the same time. It would result in large indentation and loss of space.
```

> - **Must** maintain consistency for checkboxes. 

Reviewer: ```suggestion
- **Must** maintain consistency for checkboxes. **Must not** apply checkboxes to only some items in a tree view.
```

> - **Must** maintain consistency for checkboxes. 
> - **Must** avoid checkboxes added to only some items in a tree view.

Reviewer: ```suggestion
- **Must** maintain consistency for checkboxes. **Must not** apply checkboxes to only some items in a tree view.
```

This is one line for icons and it's really the same thing stated two different ways.

> - **Should** use checkboxes to a tree view when more than one selection is allowed.

Reviewer: ```suggestion
- **Must** use checkboxes in a tree view when more than one selection is allowed.
```

There is no support for multi-selection without checkboxes.

---
category: component design guidance
state: draft
created: 2024-10-17
updated: 2024-10-22
---

> # Treeview Component

Reviewer: ```suggestion
# Tree View Design Guidance
```

Please also update the filename to match.

> The [tree view](https://clarity.design/documentation/tree-view) component presents a hierarchical list of items that have a parent-child relationship where children can be toggled into view by expanding or collapsing their parent item.

Reviewer 2: ```suggestion
The [tree view](https://clarity.design/documentation/tree-view) component displays a hierarchical list of items with a parent-child relationship, allowing users to toggle children into view by expanding or collapsing their parent item.
```

## Guidance

> - **Must** use a tree view to display and work with large amounts of hierarchical information.

Reviewer 2: ```suggestion
- Design teams **must** use a tree view to display and manage large amounts of hierarchical information, as it helps users navigate complex data structures more efficiently.
```

> - **Must not** use too many levels unless it is for user-generated information. Instead, consolidate the layers if possible.

Reviewer 2: ```suggestion
- Design teams **should not** use excessive levels in a tree view, except when displaying user-generated information. Instead, they should consolidate layers whenever possible to enhance user comprehension and streamline navigation.
```

> - **Should not** use a tree view for only one level of nested information. Use an [accordion](https://clarity.design/documentation/accordion) instead.

Reviewer 2: ```suggestion
- Design teams **should not** use a tree view for only one level of nested information. Instead, they should use an [accordion](https://clarity.design/documentation/accordion) to present the content more effectively and enhance user experience.
```

## Icon Tree

> - **Must** maintain consistency for icons. If icons need to be added to a tree view, apply the icons on all items. **Must not** apply icons to only some items in a tree view.

Reviewer 2: ```suggestion
- Design teams **must** maintain consistency in icon usage within a tree view. If icons are added, they should be applied to all items to ensure a cohesive experience. Design teams **must not** apply icons to only some items in a tree view, as this can create confusion and disrupt the visual hierarchy.
```

> - **Should** use neutral colors to avoid overloading the tree and related components with colors and shapes.

Reviewer 2: ```suggestion
- Design teams **should** use neutral colors in a tree view to prevent overwhelming users with excessive colors and shapes, promoting a clearer and more focused visual presentation.
```

> - **Should** use icons in order to visually represent the type of items in a tree.

Reviewer 2: ```suggestion
- Design teams **should** use icons to visually represent the type of items in a tree view, helping users quickly identify each item's nature and improving their ability to navigate the content effectively.
```

## Checkbox Tree

> - **Must not** use icons and checkboxes at the same time. It would result in large indentation and loss of space.  

Reviewer 2: ```suggestion
- Design teams must not use icons and checkboxes simultaneously in a tree view, as this can lead to excessive indentation and wasted space, negatively impacting usability and clarity. This approach also creates accessibility challenges, making it harder for all users, including those with visual impairments, to navigate and interact with the content effectively.
```

- **Must** maintain consistency for checkboxes. **Must not** apply checkboxes to only some items in a tree view.  

> - **Must** use checkboxes in a tree view when more than one selection is allowed.  

Reviewer 2: ```suggestion
- Design teams **must** use checkboxes in a tree view when multiple selections are allowed, enabling users to select more than one item efficiently and intuitively. 
```
